### ISSUE TITLE
[Feature Request]: Facing the situation where Chinese users are unable to use Git, Huggingface, and Pipy normally. I hope to add proxy functionality.

### ISSUE THREAD

**Issue by Ryden-kai** [Comfy-Org/desktop#450]
### Prerequisites

- [X] I have searched the existing issues and checked the recent builds/commits
- [X] I have read the documentation and this feature is not already available
- [X] This is a single feature request (not multiple features combined)

### Problem Statement

Pain points for Chinese users:
1. I hope only ComfyUI can use the proxy. Other programs on the computer are using the Chinese network normally.
2. Environment variables cannot be changed. Changing environment roll variables can be done using a mirrored URL.

### Proposed Solution

You can choose the following ways to implement it:
1. Add an environment variable setting function, allowing users to freely set environment variables. Give ComfyUI a higher degree of freedom.
2. Use the terminal to run the agent. For example: ComfyUI.exe -- proxy http://127.0.0.1:7890
![image](https://github.com/user-attachments/assets/3526cb86-5a30-44ae-8ec5-0c04f44e5a9e)
4. Add a proxy button and proxy link to the ComfyUI interface, and remind users to restart ComfyUI after enabling or disabling the proxy button to have an effect.

### Potential workflow

1. Determine proxy link
2. Enable or disable proxy

### Impact Level

Critical (Blocking my work)

**Comment by robinjhuang** [comment]
Makes sense! We are planning to add this. Are there any other features that would be helpful for users in China? Mirror urls etc?

**Comment by LiJT** [comment]
Yes need this function so bad

**Comment by zhaoqi571436204** [comment]
It is very important. In China, installation is a problem without mirrors.

**Comment by teward** [comment]
> Makes sense! We are planning to add this. Are there any other features that would be helpful for users in China? Mirror urls etc?

I'm curious, wouldn't setting system proxy be sufficient?  Is there a direct need to code the proxy functionality into Comfy?

**Comment by Ryden-kai** [comment]
> Makes sense! We are planning to add this. Are there any other features that would be helpful for users in China? Mirror urls etc?
Sure, on behalf of Chinese users, I would like to thank the developers of Comfy Org/desktop.
The following are the relevant mirror source websites in China
1 huggingface： https://hf-mirror.com/
2 pip ：  https://mirrors.tuna.tsinghua.edu.cn/pypi/web/simple
Others hope that Chinese users can see and help supplement it

**Comment by Ryden-kai** [comment]
> > Makes sense! We are planning to add this. Are there any other features that would be helpful for users in China? Mirror urls etc?
> 
> I'm curious, wouldn't setting system proxy be sufficient? Is there a direct need to code the proxy functionality into Comfy?
I hope that only Comfyui uses a proxy network, while the system and other software use the Chinese network. Some other software cannot use networks outside of China

**Comment by robinjhuang** [comment]
@Ryden-kai @zhaoqi571436204 @LiJT Tracking this issue here: 

https://github.com/Comfy-Org/desktop/issues/573

**Comment by robinjhuang** [comment]
Are you aware of any mirror sites that contain Github Releases? https://gitcode.com/indygreg/python-build-standalone This does not contain the actual Github Releases artifacts which we need to install Python

**Comment by Ryden-kai** [comment]
> Are you aware of any mirror sites that contain Github Releases? https://gitcode.com/indygreg/python-build-standalone This does not contain the actual Github Releases artifacts which we need to install Python

This is a mirror site for cloning GitHub. https://gitcode.com/gh_mirrors/co/ComfyUI.git

**Comment by Ryden-kai** [comment]
> Are you aware of any mirror sites that contain Github Releases? https://gitcode.com/indygreg/python-build-standalone This does not contain the actual Github Releases artifacts which we need to install Python

Github releases currently do not have mirror sites in China

**Comment by Ryden-kai** [comment]
> Are you aware of any mirror sites that contain Github Releases? https://gitcode.com/indygreg/python-build-standalone This does not contain the actual Github Releases artifacts which we need to install Python

You can go to this mirror site to download and install Python. https://mirrors.aliyun.com/python-release/

**Comment by Ryden-kai** [comment]
> Are you aware of any mirror sites that contain Github Releases? https://gitcode.com/indygreg/python-build-standalone This does not contain the actual Github Releases artifacts which we need to install Python

You can try downloading and installing Python from this mirror site. https://mirrors.aliyun.com/python-release/
